### PR TITLE
Select time to remember password in dialog

### PR DESCRIPTION
**Pull request by Valodim**
Using different caching times at different times is quite the valid use case. If I know I'll be doing a lot of keysigning or mail reading in the next hour or so, for example, it should be in the few hours range. Otherwise, couple minutes is fine. But it really varies.

Another advantage is that it raises awareness that passwords are cached. Users are sometimes confused why they aren't asked their password on a second operation, if they havn't noticed the cache notification before.

### CODE REVIEW

**Comment by Valodim** [comment]
Screenshots of the current implementation:





I gotta say, I don't like it much. But I'm not sure how to improve the layout either.. I thought about something like a horizontal wheel picker type thing? Or maybe just have a `+` and `-` button next to the word? Ideas would be appreciated

**Comment by dschuermann** [comment]
I like it :)
What about remembering the last selection for the next time the dialog is shown?

**Comment by Valodim** [comment]
Yeah. Either that, or leave the default as a preference. I also considered changing "Five Minutes" to "Just Now" and remember the password until the next `Intent.ACTION_SCREEN_OFF` broadcast intent. Also, maybe change the granularity to One Hour/Day/Week? Really difficult to tell which granularity makes sense...

**Comment by dschuermann** [comment]
Maybe leaving the default as a preference makes more sense in this case. It's more an exception than the rule to use long caching times. Remembering until screen goes off would be nice. I don't a specific opinion regarding granularity...

**Comment by Valodim** [comment]
I would assume that most users have a notion of what constitutes short term and long term for them, but don't need more than two or three options. I'm thinking of a preference menu with a "pick two (or three?), and one default" semantic, and the user ends up with his chosen options plus "Just Now" in the dropdown. This would also make it relatively easy to try to reduce the set of default options, but also allow flexibility to the users

**Comment by dschuermann** [comment]
Sounds great

**Comment by Valodim** [comment]
created a settings activity with the above choice flow:

**Comment by Valodim** [comment]
Q: in the current implementation, we "refresh" the passphrases when they are used. should we still do this?

I find this problematic for a couple of reasons: firstly, we can't even do it consistently anyways, for instance, apps shouldn't be able to refresh all the time to artificially extend the duration. secondly, this makes the durations quite dynamic and intransparent for the user - he picked one hour, why is this still in there after two?

**Comment by Valodim** [comment]
added "until screen off" option.

I wonder if using a shorter cache time "than usual" is actually a valid use case. As in, do users ever choose "this time, cache *shorter* than usual"? If it isn't, we can ditch the default setting altogether and use the first as default

**Comment by dschuermann** [comment]
Hmmm yeah, maybe using the first one as default is good enough. People who really prefer long cache times should just deactivate short times in the preferences.

**Comment by Valodim** [comment]
removed the default option. this might be good to go now :+1:

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/service/PassphraseCacheService.java [hunk 1]
```diff
@@ -285,10 +284,12 @@ private Passphrase getCachedPassphraseImpl(long masterKeyId, long subKeyId) thro
 
         }
 
+        // TODO do we still want this?
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/service/PassphraseCacheService.java]: no we don't want this any more. Remove it.

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/SettingsCacheTTLActivity.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2010-2014 Thialfihar <thi@thialfihar.org>
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/SettingsCacheTTLActivity.java]: wrong copyright header

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/SettingsCacheTTLFragment.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2012-2015 Dominik Schürmann <dominik@dominikschuermann.de>
+ * Copyright (C) 2015 Adithya Abraham Philip <adithyaphilip@gmail.com>
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/SettingsCacheTTLFragment.java]: wrong copyright header

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/widget/CacheTTLSpinner.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2014 Dominik Schürmann <dominik@dominikschuermann.de>
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/widget/CacheTTLSpinner.java]: wrong copyright header

File: OpenKeychain/src/main/res/layout/import_yubikey_new.xml [hunk 5]
```diff
@@ -0,0 +1,241 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dschuermann) [inline OpenKeychain/src/main/res/layout/import_yubikey_new.xml]: Why has this file been added?

**Comment by dschuermann** [comment]
I did a small review, looks great! Just some minor comments.

**Pull request merged**